src/wp-includes/formatting.php

```diff
@@ -5978,8 +5978,18 @@ function _print_emoji_detection_script() {
 	}
 
 	wp_print_inline_script_tag(
-		sprintf( 'window._wpemojiSettings = %s;', wp_json_encode( $settings ) ) . "\n" .
-			file_get_contents( ABSPATH . WPINC . '/js/wp-emoji-loader' . wp_scripts_get_suffix() . '.js' )
+		wp_json_encode( $settings, JSON_HEX_TAG | JSON_UNESCAPED_SLASHES ),
+		array(
+			'id'   => 'wp-emoji-settings',
+			'type' => 'application/json',
+		)
+	);
+
+	wp_print_inline_script_tag(
+		file_get_contents( ABSPATH . WPINC . '/js/wp-emoji-loader' . wp_scripts_get_suffix() . '.js' ),
```

Comment:
Shall we append `"\n//# sourceURL=…"` here?

Reply:
Yes. I had drafted that as part of https://github.com/WordPress/wordpress-develop/pull/9955 

See https://github.com/WordPress/wordpress-develop/pull/9955#discussion_r2389763634

But it makes sense to add it here since this PR is directly about this.

Reply:
Added in 2bc812d22cfef7cd2b321255259dd88ccd9478ca